[BUG] test_hash_pivot_groupby_nan_fallback failed on Dataproc

**Describe the bug**

Build Id: rapids_it-Dataproc/178

FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_hash_pivot_groupby_nan_fallback[[('a', RepeatSeq(Float)), ('b', Float), ('c', Long)]][IGNORE_ORDER({'local': True}), INCOMPAT, APPROXIMATE_FLOAT, ALLOW_NON_GPU(HashAggregateExec,PivotFirst,AggregateExpression,Alias,GetArrayItem,Literal,ShuffleExchangeExec,HashPartitioning,KnownFloatingPointNormalized,NormalizeNaNAndZero)]
FAILED integration_tests/src/main/python/hash_aggregate_test.py::test_struct_groupby_count[Struct(not_null)(('a', Struct(('aa', Integer),('ab', Integer))))][ALLOW_NON_GPU(ANY)]
FAILED integration_tests/src/main/python/repart_test.py::test_round_robin_sort_fallback[[('a', Array(String))]][IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(ShuffleExchangeExec,RoundRobinPartitioning)]

 =================================== FAILURES ===================================
   test_hash_pivot_groupby_nan_fallback[[('a', RepeatSeq(Float)), ('b', Float), ('c', Long)]] 
  
  data_gen = [('a', RepeatSeq(Float)), ('b', Float), ('c', Long)]
  
      @approximate_float
      @ignore_order(local=True)
      @allow_non_gpu(HashAggregateExec', PivotFirst', AggregateExpression', Alias', GetArrayItem',
              Literal', ShuffleExchangeExec', HashPartitioning', KnownFloatingPointNormalized',
              NormalizeNaNAndZero')
      @incompat
      @pytest.mark.parametrize(data_gen', [_grpkey_floats_with_nulls_and_nans], ids=idfn)
      def test_hash_pivot_groupby_nan_fallback(data_gen):
  >       assert_gpu_fallback_collect(
              lambda spark: gen_df(spark, data_gen, length=100)
                  .groupby(a')
                  .pivot(b')
                  .agg(f.sum(c')),
              "PivotFirst",
              conf=_nans_float_conf)
  
  integration_tests/src/main/python/hash_aggregate_test.py:293:

More details on the failure:
```
06:40:46  >                   [94mraise[39;49;00m Py4JJavaError(
06:40:46                          [33m"[39;49;00m[33mAn error occurred while calling [39;49;00m[33m{0}[39;49;00m[33m{1}[39;49;00m[33m{2}[39;49;00m[33m.[39;49;00m[33m\n[39;49;00m[33m"[39;49;00m.
06:40:46                          [96mformat[39;49;00m(target_id, [33m"[39;49;00m[33m.[39;49;00m[33m"[39;49;00m, name), value)
06:40:46  [1m[31mE                   py4j.protocol.Py4JJavaError: An error occurred while calling z:com.nvidia.spark.rapids.ExecutionPlanCaptureCallback.assertCapturedAndGpuFellBack.[0m
06:40:46  [1m[31mE                   : java.lang.AssertionError: assertion failed: Could not find PivotFirst in the GPU plan[0m
06:40:46  [1m[31mE                   GpuColumnarToRow false[0m
06:40:46  [1m[31mE                   +- GpuSort [b#382774 ASC NULLS FIRST], true, com.nvidia.spark.rapids.OutOfCoreSort$@240d28b6[0m
06:40:46  [1m[31mE                      +- GpuGlobalLimit 10001[0m
06:40:46  [1m[31mE                         +- GpuShuffleCoalesce 2147483647[0m
06:40:46  [1m[31mE                            +- GpuColumnarExchange gpusinglepartitioning$(), ENSURE_REQUIREMENTS, [id=#252515][0m
06:40:46  [1m[31mE                               +- GpuLocalLimit 10001[0m
06:40:46  [1m[31mE                                  +- GpuHashAggregate(keys=[b#382774], functions=[], output=[b#382774])[0m
06:40:46  [1m[31mE                                     +- GpuShuffleCoalesce 2147483647[0m
06:40:46  [1m[31mE                                        +- GpuColumnarExchange gpuhashpartitioning(b#382774, 48), ENSURE_REQUIREMENTS, [id=#252511][0m
06:40:46  [1m[31mE                                           +- GpuHashAggregate(keys=[gpuknownfloatingpointnormalized(gpunormalizenanandzero(b#382774)) AS b#382774], functions=[], output=[b#382774])[0m
06:40:46  [1m[31mE                                              +- GpuProject [b#382774][0m
06:40:46  [1m[31mE                                                 +- GpuRowToColumnar TargetSize(2147483647)[0m
06:40:46  [1m[31mE                                                    +- *(1) Scan ExistingRDD[a#382773,b#382774,c#382775L][0m
06:40:46  [1m[31mE                   [0m
06:40:46  [1m[31mE                   	at scala.Predef$.assert(Predef.scala:223)[0m
06:40:46  [1m[31mE                   	at com.nvidia.spark.rapids.ExecutionPlanCaptureCallback$.assertDidFallBack(Plugin.scala:318)[0m
06:40:46  [1m[31mE                   	at com.nvidia.spark.rapids.ExecutionPlanCaptureCallback$.assertCapturedAndGpuFellBack(Plugin.scala:312)[0m
06:40:46  [1m[31mE                   	at com.nvidia.spark.rapids.ExecutionPlanCaptureCallback.assertCapturedAndGpuFellBack(Plugin.scala)[0m
06:40:46  [1m[31mE                   	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)[0m
06:40:46  [1m[31mE                   	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)[0m
06:40:46  [1m[31mE                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[0m
06:40:46  [1m[31mE                   	at java.lang.reflect.Method.invoke(Method.java:498)[0m
06:40:46  [1m[31mE                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)[0m
06:40:46  [1m[31mE                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)[0m
06:40:46  [1m[31mE                   	at py4j.Gateway.invoke(Gateway.java:282)[0m
06:40:46  [1m[31mE                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)[0m
06:40:46  [1m[31mE                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)[0m
06:40:46  [1m[31mE                   	at py4j.GatewayConnection.run(GatewayConnection.java:238)[0m
06:40:46  [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
```

Looks like the test is expecting Pivot to be in the plan and see the fallback to the CPU, but the plan doesn't appear to have even GpuPivot in it.  cc: @nartal1

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -20,7 +20,7 @@ from functools import reduce
 from pyspark.sql.types import *
 from marks import *
 import pyspark.sql.functions as f
-from spark_session import is_before_spark_311
+from spark_session import is_before_spark_311, with_cpu_session
 
 _no_nans_float_conf = {'spark.rapids.sql.variableFloatAgg.enabled': 'true',
                        'spark.rapids.sql.hasNans': 'false',
@@ -281,7 +281,6 @@ def test_hash_grpby_pivot(data_gen, conf):
             .agg(f.sum('c')),
         conf=conf)
 
-
 @approximate_float
 @ignore_order(local=True)
 @allow_non_gpu('HashAggregateExec', 'PivotFirst', 'AggregateExpression', 'Alias', 'GetArrayItem',
@@ -290,10 +289,19 @@ def test_hash_grpby_pivot(data_gen, conf):
 @incompat
 @pytest.mark.parametrize('data_gen', [_grpkey_floats_with_nulls_and_nans], ids=idfn)
 def test_hash_pivot_groupby_nan_fallback(data_gen):
+    # collect values to pivot in previous, to avoid this preparation job being captured
+    def fetch_pivot_values(spark):
+        max_values = spark._jsparkSession.sessionState().conf().dataFramePivotMaxValues()
+        df = gen_df(spark, data_gen, length=100)
+        df = df.select('b').distinct().limit(max_values + 1).sort('b')
+        return [row[0] for row in df.collect()]
+
+    pivot_values = with_cpu_session(fetch_pivot_values)
+
     assert_gpu_fallback_collect(
         lambda spark: gen_df(spark, data_gen, length=100)
             .groupby('a')
-            .pivot('b')
+            .pivot('b', pivot_values)
             .agg(f.sum('c')),
         "PivotFirst",
         conf=_nans_float_conf)
@@ -333,10 +341,19 @@ def test_hash_multiple_grpby_pivot(data_gen, conf):
 @incompat
 @pytest.mark.parametrize('data_gen', [_grpkey_floats_with_nulls_and_nans], ids=idfn)
 def test_hash_pivot_reduction_nan_fallback(data_gen):
+    # collect values to pivot in previous, to avoid this preparation job being captured
+    def fetch_pivot_values(spark):
+        max_values = spark._jsparkSession.sessionState().conf().dataFramePivotMaxValues()
+        df = gen_df(spark, data_gen, length=100)
+        df = df.select('b').distinct().limit(max_values + 1).sort('b')
+        return [row[0] for row in df.collect()]
+
+    pivot_values = with_cpu_session(fetch_pivot_values)
+
     assert_gpu_fallback_collect(
         lambda spark: gen_df(spark, data_gen, length=100)
             .groupby()
-            .pivot('b')
+            .pivot('b', pivot_values)
             .agg(f.sum('c')),
         "PivotFirst",
         conf=_nans_float_conf)

